src/sources/exec.rs

```diff
@@ -0,0 +1,868 @@
+) -> Option<Vec<Event>> {
+    let mut command = build_command(config);
+
+    // Mark the start time just before spawning the process as
+    // this seems to be the best approximation of exec duration
+    let now = Instant::now();
+    let child = command.spawn();
+
+    match child {
+        Ok(child) => {
+            let schedule = Duration::from_secs(exec_interval_secs);
+            let pid = child.id();
+
+            // Wait for our task to finish, wrapping it in a timeout
+            let timeout = tokio::time::timeout(schedule, child.wait_with_output());
+            let timeout_result = timeout.await;
+
+            match timeout_result {
+                Ok(output) => match output {
+                    Ok(output) => {
+                        let mut events = Vec::new();
+                        let elapsed_millis = now.elapsed().as_millis();
+                        let exit_status = output.status.code();
+
+                        let stdout_events = process_scheduled_output(
+                            config,
+                            event_per_line,
+                            exec_duration_millis_key.clone(),
+                            pid,
+                            exit_status,
+                            elapsed_millis,
+                            output.stdout,
+                            STDOUT,
+                        );
+
+                        if let Some(mut stdout_events) = stdout_events {
+                            events.append(&mut stdout_events);
+                        }
+
+                        // Optionally include stderr
+                        if config.include_stderr.unwrap_or(true) {
+                            let stderr_events = process_scheduled_output(
+                                config,
+                                event_per_line,
+                                exec_duration_millis_key.clone(),
+                                pid,
+                                exit_status,
+                                elapsed_millis,
+                                output.stderr,
+                                STDERR,
+                            );
+
+                            if let Some(mut stderr_events) = stderr_events {
+                                events.append(&mut stderr_events);
+                            }
+                        }
+
+                        if events.is_empty() {
+                            None
+                        } else {
+                            Some(events)
+                        }
+                    }
+                    Err(command_error) => {
+                        error!("Error during scheduled exec run.");
+                        emit!(ExecFailed {
+                            error: command_error
+                        });
+                        None
+                    }
+                },
+                Err(_) => {
+                    error!("Timed out during scheduled exec run.");
```

Comment:
The internal event should already emit a log so I don't think we need this one too.

Reply:
Removed